### DISCUSSION TITLE
Alternative accept with a different outcome?

### DISCUSSION THREAD

**Discussion by 097115** [davatorium/rofi#1851] [Q&A] [answered]
Hi!

When scripting with `fzf`, for instance, it's possible to bind a key to accept an entry in such a way that the script will be able to distinguish later on *how* was the entry selected (e.g., via `fzf --bind='ctrl-x:execute(echo -n ctrlx:)+accept'`). Which would make possible to program different actions for a selected entry depending on the way it was accepted.

As I understand, there are no similar workarounds in `rofi`, right? Just wanted to clarify :) (I'm using it in the `-dmenu` mode).

Thanks :).

**Comment by DaveDavenport** [comment]
https://davatorium.github.io/rofi/current/rofi-dmenu.5/#return-value ?

**Comment by 097115** [comment]
@DaveDavenport, thank you for the link!

Indeed, it mentions some custom bindings, which, as I understand, are `ctrl-return` and `shift-return`, right?

Thus, I tried the following example:

    echo 'a|b|c' | rofi -dmenu -sep '|' 

First, I pressed `<ctrl-return>`, it returned an empty string for some reason, and the exit code was 0. Next, I re-run the example and pressed `<shift-return>`, which indeed returned the selected row, but the exit code was 0 again. I also tried `<super-1>`, for example, but the exit code was still 0.

Obviously, I missing something here, but what? :) Or may be my `rofi` is just too old? It's 1.7.1 @ Ubuntu 22.04.3.

Would be grateful for a hint, and thanks again :)

**Reply by DaveDavenport** [reply, marked as answer]
> 10-28: Row accepted by custom keybinding.

You can specify up to 19 custom keybindings that you can see from the script.

See https://davatorium.github.io/rofi/current/rofi-keys.5/#kb-custom-1 for the list of default bindings (you can change them, see same manpage).

To see an example, check out rofi-theme-selector that is installed together with rofi.

**Reply by DaveDavenport** [reply]
Alt-a is set here:
* https://github.com/davatorium/rofi/blob/next/script/rofi-theme-selector#L145
And detected here:
* https://github.com/davatorium/rofi/blob/next/script/rofi-theme-selector#L165

**Reply by 097115** [reply]
Thanks a lot, Dave, and sorry for my blindness.

**Reply by DaveDavenport** [reply]
no problem, just very busy, so answers  are short :smile:
